python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -707,15 +730,15 @@ class RandomForestClassifier(Base):
 
         fil_model = ForestInference()
         tl_to_fil_model = \
-            fil_model.load_from_randomforest(treelite_handle,
+            fil_model.load_from_randomforest(self.treelite_handle,
                                              output_class=output_class,
                                              threshold=threshold,
                                              algo=algo,
                                              storage_type=storage_type)
 
         preds = tl_to_fil_model.predict(X, output_type=out_type,
                                         predict_proba=predict_proba)
-        tl.free_treelite_model(treelite_handle)
```

Comment:
now the self.treelite_handle is freed in the __del__ destructor for the instance. That way we can cache it and use it again if we call predict multiple times, but it will be cleaned up when the instance is done.